indigo_drivers/agent_scripting/library/Sequencer.js

```diff
@@ -248,36 +254,61 @@ var indigo_flipper = {
 		"Server"
 	],
 
-	waiting_for_transit: false,
-	waiting_for_slew: false,
+	// waiting_for_transit: false,
+	// waiting_for_slew: false,
 	waiting_for_sync_and_center: false,
 	waiting_for_guiding: false,
 	resume_guiding: false,
 	use_solver: false,
+	// transitTimer,
+	// transitTimerSet: false,
+
+
+	transitTimerTrigger: function() {
+		indigo_send_message("Meridian flip started");
+		waiting_for_transit = false;
+		indigo_send_message("Setting slew flag");
+		waiting_for_slew = true;
+		indigo_send_message("Starting slew");
+		indigo_change_switch_property("Mount Agent", "AGENT_START_PROCESS", { SLEW: true});
+		indigo_send_message("Cancelling transit timer");
+		indigo_cancel_timer(transitTimer);
+		indigo_send_message("Unsetting transit timer flag");
+		transitTimerSet = false;
+	},
 
 	on_update: function(property) {
-		if (this.waiting_for_transit)
+		if (waiting_for_transit)
 			indigo_log("waiting_for_transit " + property.name + " -> " + property.state);
-		else if (this.waiting_for_slew)
+		else if (waiting_for_slew)
 			indigo_log("waiting_for_slew " + property.name + " -> " + property.state);
 		else if (this.waiting_for_sync_and_center)
 			indigo_log("waiting_for_sync_and_center " + property.name + " -> " + property.state);
 		else if (this.waiting_for_guiding)
 			indigo_log("waiting_for_guiding " + property.name + " -> " + property.state);
-		if (this.waiting_for_transit && property.device == this.devices[3] && property.name == "AGENT_MOUNT_DISPLAY_COORDINATES_PROPERTY") {
+		if (waiting_for_transit && property.device == this.devices[3] && property.name == "AGENT_MOUNT_DISPLAY_COORDINATES_PROPERTY") {
+			if (!transitTimerSet) {
+				transitTimerSet = true;
+				// Convert TIME_TO_TRANSIT to seconds and set a timer for that long
+				var interval = property.items.TIME_TO_TRANSIT*3600 + 60;
+				indigo_send_message("Meridian flip timer set for " + String(interval) + " seconds");
+				transitTimer = indigo_set_timer(this.transitTimerTrigger, interval);
```

Comment:
Is it technically possible to create a timer using an object method rather than a global function? I wasn't able to access the member attributes so moved the variables needed here to global scope.

Reply:
Do you really need it? With the trick mentioned above you can trigger flip before transit. What may be needed in such case is some delay, because I'm not sure if the mount will actually do the flip if you will slew 10s before transit.

Reply:
I tried that trick and the issue with that is if I trigger the flip before transit so the mount doesn't run into meridian limits, the flip doesn't actually happen like you mention.

But in general, I feel like if you pause the batch a certain amount of time before transit and you know how much time is remaining for transit it would be better to just wait for that amount of time plus some buffer before triggering a meridian flip. It would also allow stopping tracking when the batch is paused and restarting tracking once meridian flip is complete so the mount never has the chance to hit meridian limits which would be especially useful in some mounts that might not be able to recover automatically from a meridian limit violation.